### ISSUE TITLE
Is it okay to train the model with original size?

### ISSUE THREAD

**Issue by saikrishnadas** [AlexeyAB/darknet#6925]
I have some images of 1270 x 780p, it is okay to train those images without croping to 416x416(my yolo cfg) ?
Darknet does all the crop by itself?

And i came across darknet use `letterbox_image` aspect ratio. Is it automatically implemented while training or should i change the code to make that aspect ratio work ?

@AlexeyAB

**Comment by stephanecharette** [comment]
I don't believe the letterbox option is used anymore.  There are some tickets in github that document things that don't work anymore when people try to enable it.

Darknet doesn't care about the image size.  It will resize all images to match the network size.  What you need to do is make sure that your annotations in the image aren't so small that when resized to the network dimensions they would be less than the minimum size of 16x16.

**Comment by saikrishnadas** [comment]
Darknet resizing my image, will it crop out the image? Lossing all the objects that are present at the wide end of the image?

**Comment by stephanecharette** [comment]
No, the image is stretched/resized, not cropped.  That is why all the coordinates are normalized 0.0 to 1.0.  This way it doesn't matter what size image is used, or how the image is resized when the network dimensions are changed.

**Comment by saikrishnadas** [comment]
So, does it affect while testing ? I mean my test images comes in original shape. If the image is stretched, eventually the object inside the image changes its shape right ?

**Comment by stephanecharette** [comment]
Remember that it changes "shape" during training as well.  So in the end, it cancels out.  As long as your training images are similar to what you'll be using.

**Comment by saikrishnadas** [comment]
Yeah, training images get stretched. Totally understood.
But my images that im going to use in real-time will be in original shape ( no stretching ). Does the that affect my model , as my model was trained on stretched objects. @stephanecharette

**Comment by stephanecharette** [comment]
They will be stretched.  Darknet stretches all images, not just training ones.
